### ISSUE TITLE
Ingest migration for dotted fields to LS field references

### ISSUE THREAD

**Issue by suyograo** [elastic/logstash#7294]
One of the tests I tried after merging https://github.com/elastic/logstash/pull/7275 was a date processor with dots in `target_field`

```
{
  "description" : "...",
  "processors" : [
    {
      "date" : {
        "field" : "initial_date",
        "target_field" : "apache.timestamp",
        "formats" : ["dd/MM/yyyy hh:mm:ss", "dd/MM/yyyy"],
        "timezone" : "Europe/Amsterdam",
        "locale": "en"
      }
    }
  ]
}
```

LS

```
filter {
   date {
      match => [
         "initial_date",
         "dd/MM/yyyy hh:mm:ss",
         "dd/MM/yyyy"
      ]
      target => "apache][timestamp"
      timezone => "Europe/Amsterdam"
      locale => "en"
   }
}
```

The result is not correct: `target => "apache][timestamp"`

@

**Comment by suyograo** [comment]
@original-brownbear I use `dots_to_square_brackets()` for this, but seems to be missing something. Can you take a look please?

**Comment by original-brownbear** [comment]
@suyograo sure, just at it with the drying up anyways. WIll handle this :)

**Comment by original-brownbear** [comment]
@suyograo fixed in https://github.com/original-brownbear/logstash/tree/dry-up-js will clean this up properly after a little sleep :)

**Comment by original-brownbear** [comment]
@suyograo there we go https://github.com/elastic/logstash/pull/7296 :)

**Comment by original-brownbear** [comment]
already fixed in #7296
